specification/chaos/resource-manager/Microsoft.Chaos/preview/2023-01-31-preview/types/applications.json

```diff
@@ -0,0 +1,360 @@
+          "items": {
+            "$ref": "#/definitions/faultCapabilityProperties"
+          },
+          "description": "faults supported by torch for the function"
+        }
+      },
+      "description": "Application function in which fault can be injected",
+      "required": [
+        "signature"
+      ]
+    },
+    "faultCapabilityProperties": {
+      "type": "object",
+      "properties": {
+        "faultUrn": {
+          "type": "string",
+          "description": "URN for fault capability"
+        },
+        "parameters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/faultParameter"
+          },
+          "description": "Input Parameters supported by a fault"
+        }
+      },
+      "description": "Capabilities supported by the faults"
+    },
+    "faultParameter": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "name of a fault parameter"
+        }
+      }
+    },
+    "sourceProperties": {
+      "type": "object",
+      "properties": {
+        "codePath": {
+          "type": "string",
+          "description": "code path where function is defined"
+        },
+        "startLine": {
+          "type": "integer",
+          "format": "int32",
+          "description": "line in the code where function starts"
+        },
+        "startColumn": {
+          "type": "integer",
+          "format": "int32",
+          "description": "column in the code where function starts"
+        },
+        "endLine": {
+          "type": "integer",
+          "format": "int32",
+          "description": "line in the code where function ends"
+        },
+        "endColumn": {
+          "type": "integer",
+          "format": "int32",
+          "description": "column in the code where function ends"
+        }
+      },
+      "description": "defines the function code path and location in the class"
+    },
+    "functionArguments": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Type of a function argument"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of a function argument"
+        }
+      }
+    },
+    "signatureProperties": {
+      "type": "object",
+      "properties": {
+        "functionNamespace": {
+          "type": "string",
+          "description": "Namespace in which the function is defined"
+        },
+        "type": {
+          "type": "string",
+          "description": "ClassName in which the function is defined"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of a function"
+        },
+        "genericTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Types defined if Class is a generic class"
+        },
+        "genericCount": {
```

Comment:
>genericCount [](http://example.com/codeflow?start=9&length=12)

Why include a count for the adjacent array object which has its own length? I'd leave this out.

Reply:
To clarify, this is my main blocking feedback at this point, if you need it for backcompat or something, let me know.

Reply:
As of now our data publisher id publishing the count of generic separately and its not user defined. So would like to keep the count as of now. In future versions we should be able to deprecate it .

Reply:
I'm still not feeling clear why you don't just remove it now, filter it out, or at least mark it deprecated NOW, if you know you plan to remove it

Reply:
(as a signal to users they shouldn't depend on it)